### PR TITLE
fix: always prioritize reg or index flags

### PR DESCRIPTION
**Pull request by hi-rustin**

### CODE REVIEW

**Comment by hi-rustin** [comment]
I will add a test for it later.

**Comment by hi-rustin** [comment]
@epage Ant thoughts about the priority of the source ID here? Right now, we prefer the source ID from the lock file.

**Comment by epage** [comment]
Could you add more detail on the problem this is trying to address?

**Comment by hi-rustin** [comment]
> Could you add more detail on the problem this is trying to address?

For instance, suppose you have a dependency from a private registry, and you use the `--registry crates-io` command. In such a case, we will prioritize the private registry over crates-io. We may need to add a warning to inform users about this behavior. If the package exists in your workspace, we will retrieve it from that private registry instead of the registry you specified.

**Comment by epage** [comment]
> For instance, suppose you have a dependency from a private registry, and you use the --registry crates-io command. In such a case, we will prioritize the private registry over crates-io. We may need to add a warning to inform users about this behavior. If the package exists in your workspace, we will retrieve it from that private registry instead of the registry you specified.

imo we should prioritize the command-line argument and and no warning should be given.  The user told us they intended to do something and we should respect that.

File: src/ops/info.rs [hunk 1]
```diff
@@ -35,7 +35,11 @@ pub fn info(spec: &str, config: &Config, reg_or_index: Option<RegistryOrIndex>)
         }
     }
 
-    let source_ids = get_source_id(config, reg_or_index, package_id)?;
+    let (use_package_source_id, source_ids) = get_source_id(config, reg_or_index, package_id)?;
+    // If we don't use the package's source, we need to query the package ID from the specified registry.
+    if !use_package_source_id {
+        package_id = None;
+    }
```
Comment (epage) [inline src/ops/info.rs:42]: Can we have a test for this?

**Review by epage — APPROVED** [review]
Looking good.  Ideally, we do tests now so the work to upstream isn't too monumental but I leave that to you.

File: src/ops/info.rs (hunk 1 above)
Comment (hi-rustin) [reply src/ops/info.rs:42]: Added.

**Review by hi-rustin — COMMENTED** [review]
🔢 Self-check

**Pull request merged**